### ISSUE TITLE
`jit.export` analoge for `torch.export`

### ISSUE THREAD

**Issue by randolf-scholz** [pytorch/pytorch#167631]
### 🚀 The feature, motivation and pitch

According to the documentation, [`TorchScript` is deprecated in favor of `torch.export`](https://docs.pytorch.org/docs/stable/jit.html). 

However, `torch.jit.script` offered some functionality that does not seem to be covered by `torch.export`, specifically the ability to export multiple entry points via the `@jit.export` decorator. This is useful in many situations, for instance when working with Normalizing Flows and wanting to use both forward and inverse method, probabilistic models with multiple relevant methods, or for declaring additional state-modifying functions.

Without this functionality, it's unclear how to convert models that relied on `jit.export` to the new `torch.export` setup.

Related Discussions:

- https://github.com/pytorch/executorch/issues/7458
- https://discuss.pytorch.org/t/export-multiple-functions-of-a-pytorch-module/194816

**Comment by yushangdi** [comment]
cc @zhxchen17 maybe the Exporter would help here?
